File: leetcode/1easy/오하늘/1995.py
from itertools import combinations
<<<<<<< HEAD
class Leetcode.이재훈.이재훈.Solution:
=======
class Leetcode.이재훈.Solution:
>>>>>>> ab7f46f73d8f18bcd38c5cf06ecabae64f53a424
    def countQuadruplets(nums) -> int:
        # 중복 가능
        ans = 0
        for a, b, c, d in combinations(nums, 4):
            if a+b+c == d:
                ans+=1
        return ans

File: leetcode/1easy/박은비/1995.swift
/*
4 <= nums.length <= 50
1 <= nums[i] <= 100
*/
class Leetcode.이재훈.이재훈.Solution {
    func countQuadruplets(_ nums: [Int]) -> Int {
        let n = nums.count
        var count = 0

        for a in 0..<n {
            for b in (a+1)..<n {
                for c in (b+1)..<n {
                    for d in (c+1)..<n {
                        if nums[a] + nums[b] + nums[c] == nums[d] {
                            count += 1
                        }
                    }
                }
            }
        }

        return count
    }
}
